gdx/src/com/badlogic/gdx/files/FileHandle.java

```diff
@@ -507,12 +528,23 @@ public void writeBytes (byte[] bytes, int offset, int length, boolean append) {
 		return handles;
 	}
 
-	/** Returns true if this file is a directory. Always returns false for classpath files. On Android, an
-	 * {@link FileType#Internal} handle to an empty directory will return false. On the desktop, an {@link FileType#Internal}
-	 * handle to a directory on the classpath will return false. */
+	 /**
+	  * Returns true if this file is a directory. Always returns false for classpath files. On Android, an
+	  * {@link FileType#Internal} handle to an empty directory will return false unless you are using an asset index. On desktop,
+	  * an {@link FileType#Internal} handle will always return false unless you are using an asset index.
+	  *
+	  * See {@link #list()} documentation for more information about using an Asset Index
+	  */
 	public boolean isDirectory () {
-		if (type == FileType.Classpath) return false;
-		return file().isDirectory();
+		 if (type == FileType.Classpath)
+			  return false;
+		 if (type == FileType.Internal) {
+			  for (String indexpath : assetIndex()) {
+					if (indexpath == path() + "/")

```

Comment:
This will probably [never return true](https://stackoverflow.com/questions/767372/string-equals-versus#767379) and generates a lot of garbage.

`indexpath.length() + 1 == path && indexpath.startsWith(path) && indexpath.endsWith("/")` will do the same comparison without any allocations, but caching `path() + "/"` and doing the comparison with `equals()` will also work.